spec/memo_wise_spec.rb

```diff
@@ -115,15 +115,15 @@ def private_memowise_method
         @private_memowise_method_counter += 1
         "private_memowise_method"
       end
-      memo_wise :private_memowise_method
       private :private_memowise_method
+      memo_wise :private_memowise_method
```

Comment:
The ordering here meant we were making the methods `private` or `protected` after memoizing them. We need to change their visibility _before_ memoizing them so that our memoization can preserve visibility.

I have added a note to describe this in the README in Issue #37